src/Stratis.Bitcoin/Consensus/ChainedHeaderTree.cs

```diff
@@ -1204,10 +1209,21 @@ private void CheckMaxReorgRuleViolated(ChainedHeader chainedHeader)
         }
 
         /// <inheritdoc />
-        public bool IsAtBestChainTip()
+        public bool IsConsensusConsideredToBeSynced()
         {
             this.logger.LogTrace("()");
 
+            // If the height of consensus is less than the tip margin, the node is considered to be synced.
```

Comment:
Yes you are right, this should be the opposite. However this would mean that in all the integration we would have to mine at least 5 blocks before we persist? What is youe thoughts on this?

Reply:
How so? Let's say there are four blocks on the network and the node connects to it having zero blocks. We just need to make sure that the expression below – the normal one – compares correctly:

`our height + threshold > best = 0 + 5 > 4`

=> yes => save even the first block

so instead of minus, use plus below to avoid overflow and negative numbers and it should be fine

Reply:
So I think we can remove this part of the code and instead just have a good equation down below and we should be fine

Reply:
Ok yeah, so we essentially invert the expression 👍 DONE.